pkg/bucketaccessrequest/bucketaccessrequest.go

```diff
@@ -57,13 +58,24 @@ func (b *bucketAccessRequestListener) Add(ctx context.Context, obj *v1alpha1.Buc
 	return nil
 }
 
+// Update is called in response to a change to BucketAccessRequesst. At this point
+// BucketAccess cannot be changed once created as the Provisioner might have already acted upon the create BucketAccess and created the backend Bucket Credentials
+// Changes to Protocol, Provisioner, BucketInstanceName, BucketRequest cannot be allowed. Best is to delete and recreate a new BucketAccessRequest
+// Changes to ServiceAccount, PolicyActionsConfigMapData and Parameters should be considered in lieu with sidecar implementation
 func (b *bucketAccessRequestListener) Update(ctx context.Context, old, new *v1alpha1.BucketAccessRequest) error {
 	glog.V(1).Infof("Update called for BucketAccessRequest %v", old.Name)
+	if (old.ObjectMeta.DeletionTimestamp == nil) &&
```

Comment:
Is this condition guaranteed by k8s to exist upon delete of the BAR?  Is it ever possible to miss the initial delete event but catch the following BAR update and now both timestamp != nil?

Reply:
yeah, we should just check if new.DeletionTimestamp != nil

Reply:
true, Fixed